File: output/ChatGPT/codenet/Java/Python/s443066739.py
n = int(input())
pre = [0] * (n+1)

for i in range(1, n+1):
    pre[i] = pre[i-1] + int(input())

min_val = float('inf')

for i in range(1, n):
    val1 = pre[i]
    val2 = pre[n] - pre[i]
    
    min_val = min(min_val, abs(val1 - val2))

print(min_val)

File: output/ChatGPT/codenet/Java/C++/s443066739.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    int n;
    cin >> n;
    vector<long long> pre(n+1);

    for(int i=1;i<=n;i++) {
        cin >> pre[i];
        pre[i] += pre[i-1];
    }

    long long minDiff = LLONG_MAX;

    for(int i=1;i<n;i++) {
        long long val1 = pre[i], val2 = pre[n]-pre[i];
        minDiff = min(minDiff, abs(val1-val2));
    }

    cout << minDiff << endl;
    return 0;
}